src/gar.lua

```diff
@@ -495,7 +505,7 @@ function gar.decreaseDelegateStake(gatewayAddress, delegator, qty, currentTimest
 	-- update the gateway
 	GatewayRegistry[gatewayAddress] = gateway
 	return {
-		gateway = gar.getGateway(gatewayAddress),
+		gateway = gateway,
```

Comment:
Good change. Might want to search the codebase for other opportunities to save cycles like this.

Reply:
updated other uses here - [8f13aa5](https://github.com/ar-io/ar-io-network-process/pull/161/commits/8f13aa515d4798eeb8fe2920e20814ba6df7b6d9)

there are still other uses, but they are when we are operating on the resulting gateway to increment stats, change weights, etc. and the deep clone prevents unintentionally modifying the reference gateway before all modifications are made.